go/vt/vtgate/planbuilder/operators/expressions.go

```diff
@@ -57,8 +57,24 @@ func BreakExpressionInLHSandRHS(
 	}, nil).(sqlparser.Expr)
 
 	if err != nil {
-		return nil, nil, nil, err
+		return JoinColumn{}, err
 	}
 	ctx.JoinPredicates[expr] = append(ctx.JoinPredicates[expr], rewrittenExpr)
+	col.RHSExpr = rewrittenExpr
 	return
 }
+
+// BreakExpressionInLHSandRHSOld takes an expression and
+// extracts the parts that are coming from one of the sides into `ColName`s that are needed
+func BreakExpressionInLHSandRHSOld(
```

Comment:
Why is this required? Couldn't we change the places using this to expect a JoinColumn instead?